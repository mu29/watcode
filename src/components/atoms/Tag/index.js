import styled from 'styled-components'
import { palette } from 'services/style'

const Tag = styled.div`
  display: inline-block;
  margin-right: 0.375rem;
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  font-weight: 200;
  color: ${palette('gray.60')};
  background-color: ${palette('gray.20')};
  border-radius: 0.25rem;
  white-space: nowrap;
`

export default Tag
