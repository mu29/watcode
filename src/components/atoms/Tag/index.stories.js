import React from 'react'
import { storiesOf } from '@storybook/react'
import Tag from '.'

storiesOf('Atoms/Tag', module)
  .add('기본', () => (
    <Tag>라노벨</Tag>
  ))
