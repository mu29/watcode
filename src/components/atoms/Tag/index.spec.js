import React from 'react'
import Tag from '.'

const component = (props = {}) => withTheme(<Tag { ...props } />)

describe('<Tag />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })

  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = render(component({ children: 'test' }))
    expect(wrapper).toMatchSnapshot()
  })
})
