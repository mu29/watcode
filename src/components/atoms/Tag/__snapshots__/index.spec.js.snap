// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
.c0 {
  display: inline-block;
  margin-right: 0.375rem;
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  font-weight: 200;
  color: #9E9E9E;
  background-color: #F5F5F5;
  border-radius: 0.25rem;
  white-space: nowrap;
}

<div
  class="c0"
>
  test
</div>
`;

exports[`<Tag /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: inline-block;
  margin-right: 0.375rem;
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  font-weight: 200;
  color: #9E9E9E;
  background-color: #F5F5F5;
  border-radius: 0.25rem;
  white-space: nowrap;
}

<div
  class="c0"
/>
`;
